asset/records.go

```diff
@@ -143,6 +150,13 @@ func NewLeafScriptKeyRecord(scriptKey **btcec.PublicKey) tlv.Record {
 	)
 }
 
+func NewLeafXOnlyGroupKeyRecord(groupKey **GroupKey) tlv.Record {
```

Comment:
See above, not sure we'll need this.

Reply:
I think we do want it? If we're leaving in the current logic for group key struct encode/decode in this PR.

Changing it now to pull only the key from `GroupKey` though instead of using a new field in the `Asset` struct.

Reply:
> I think we do want to change the encoding to only the x-only key though? Just not change the top-level struct.

x-only for the group key? Rationale being that it's now an actual taproot output key so it should itself be 32-bytes I take it?

Reply:
Yeah, right now we actually don't follow the BIP since we do serde for the group key as compressed 33-byte vs. Schnorr 32-byte.

https://github.com/lightninglabs/taproot-assets/blob/429b3ea2762ba5ef8a351d53454c672c2a54017b/asset/encoding.go#L557

Compared to the script key,  this key will always be 32-byte b/c it can only be derived with a TapTweak, so we could just do serde with 32 bytes instead of 33.

A separate question is whether we should store it in the DB as 32 or 33 bytes - it's 33 right now, IIRC that was for consistency with all the other keys we store?